script/install.sh

```diff
@@ -62,26 +62,43 @@ retry() {
         fi
     done
 }
-
 # url, path, [gpg signature url to verify]
+# url and the gpg signature can be an array of urls
 download() {
-    local url path
-    url=$1
-    path=$2
-
-    retry curl "$url" -o "$path"
+    local mirrors path
+    mirrors=($1)
+    path="$2"
+
+    try_all_mirrors() {
+        for i in "${!mirrors[@]}" ; do
+            if [ "$i" -gt 0 ] ; then
+                log "Falling back to mirror: ${mirrors[$i]}"
+            fi
+            if curl "${mirrors[$i]}" -o "$path" ; then
+                break
+            fi
+        done
+    }
+    retry try_all_mirrors
     if [ ! -z "${3:-}" ]; then
         verify "$3" "$path"
     fi
 }
 
-# url
+# url (can be an array of urls)
 fetch() {
-    local url path
-    url=$1
+    local mirrors path
+    mirrors=($1)
```

Comment:
This does IFS/whitespace splitting. Rather pass all the arguments separately on the call-side and use `("$@")` here.
```sh
cnt() { echo $#; }
bug() { ary=($@); cnt "${ary[@]}"; }
bug 1 2 # -> 2
bug '1 1' 2 # -> 3 :/
ok() { ary=("$@"); cnt "${ary[@]}"; }
ok 1 2 # -> 2
ok '1 1' 2 # -> 2 :)
```

Reply:
I'm aware (I did it on purpose after all) - urls aren't supposed to contain spaces :wink: 
I did have this, s.t. I can pack the urls for mirrors and their signatures as individual arguments for the `download` function.